### PR TITLE
Feat/confusion matrix

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Describe your changes
Added confusion matrix functionality, and introduced a style of adding new functionality that doesn't obfuscate functionality within a class

## Issue ticket number and link
DSP-48

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have built basic tests for new functionality (particularly new evaluators)
- [ ] If new libraries have been added, I have checked that [readthedocs](https://readthedocs.org/projects/credoai-lens/) API documentation is constructed correctly
- [ ] Will this be part of a major product update? If yes, please write one phrase about this update.

## Extra-mile Checklist
- [ ] I have thought expansively about edge cases and written tests for them

### CODE REVIEW

File: credoai/evaluators/performance.py [hunk 1]
```diff
@@ -57,11 +58,6 @@ def __init__(self, metrics=None):
         self.prob_metrics = None
         self.failed_metrics = None
 
-    def _validate_arguments(self):
-        check_existence(self.metrics, "metrics")
-        check_data_instance(self.assessment_data, TabularData)
-        check_artifact_for_nulls(self.assessment_data, "Data")
-
```
Comment (fabrizio-credo) [inline credoai/evaluators/performance.py:64]: The evaluator schema I formalized in the docs is:
- init()
- validate()
- setup()

This mirrors what happens in the __call__() method. Can you move validate up before setup?

**Review by fabrizio-credo — APPROVED** [review]
Just one minor item on the identity_verification evaluator to take care of.

Feel free to merge after.
